dvc/hash_info.py

```diff
@@ -7,6 +8,17 @@
 DirInfo = Dict[str, str]
 
 
+class HashName(str, enum.Enum):
+    MD5 = "md5"  # Raw MD5
+    MD5_D2U = "md5-d2u"  # DVC dos2unix MD5
```

Comment:
As it's written right now, for users who have explicitly opted into using dos2unix, we will write a config file containing version `1.0`, and dvc.lock files will now contain hashes like

```
md5-d2u: abc123
```

This seems more "correct" to me versus continuing to write lock files with `md5: abc123`, but the downside would be that anyone on DVC < 2.0 won't be able to read lock files with `md5-d2u` hash names.